shared/pkg/job_manager/config.go

```diff
@@ -0,0 +1,20 @@
+// Copyright © 2022 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package job_manager // nolint:revive,stylecheck
+
+type IsConfig interface {
+	IsConfig()
+}
```

Comment:
I don't really get the reasoning behind this interface, I think we should either statically type the config with a top level struct which all the of the jobs accept and then pull out the stuff they care about, or use a *Viper instead to remove the need for a top level struct.